### DISCUSSION TITLE
how could I stack the graph with different number of node and edge_index in to a mini-batch?

### DISCUSSION THREAD

**Discussion by luciaL** [pyg-team/pytorch_geometric#2439] [Q&A]
I have thousands of graphs but with the different number of nodes and edge_index. During training, it will take a long time to stack along one dimension with Dataloader. So I'd like to try the mini-batch method. But the DenseDataLoader and cat_dim methods need the number of nodes is all the same.
`class MyData(Data): 
    def __cat_dim__(self, key, item): 
    if key == 'foo':
         return None 
    else: 
        return super().__cat_dim__(key, item)`

How could I accelerate the training? I think the first one is to batching along a new dimension without stacking. The two above methods not gonna work with the different number of nodes.
The second method is using SparseTensor, but this works with batch_size =1.

**Comment by rusty1s** [comment]
If you have a dataset with graphs of varying size, I suggest you use `torch_geometric.data.DataLoader`. If you notice slow batching, I suggest you try to increase the `num_workers` argument.

**Reply by luciaL** [reply]
Thank you so much for your reply.
